lrauv_ignition_plugins/scripts/make_dem_utm_tiles.py

```diff
@@ -0,0 +1,141 @@
+# Research Institute (MBARI) and the David and Lucile Packard Foundation
+
+import subprocess
+
+import netCDF4 as nc
+import numpy as np
+from pyproj import Proj
+
+
+def mk_tiles(map_path: str, utm_zone: int, out_path: str, tile_size: int, overlap: int):
+    """
+    Generates digital elevation map (DEM) tiles from parent DEM using Generic Mapping Tool (GMT).
+
+    To install GMT,see: https://www.generic-mapping-tools.org
+
+    Tiles are generated with size of tile_size x tile_size, and with "overlap"
+    meters shared with adjacent tiles on each edge.
+
+    Tile boundaries will be computed starting at the bottom/right corner of
+    the parent DEM (rounded to nearest 100 m).
+
+    :param map_path: file path to full-resolution DEM
+    :param utm_zone: full-res DEM UTM zone
+    :param out_path: output path for tile files
+    :param tile_size: meters, each tile will have are of tile_size x tile_size
+    :param overlap: meters, tiles will have "overlap" meters shared with adjacent tiles on each edge
+    :return: void
+    """
+    unique_size = tile_size - 2 * overlap
+
+    # Open the file and create an instance of the ncCDF4 class
+    nc_fid = nc.Dataset(map_path, 'r')
+
+    # Extract data from NetCDF file
+    x = nc_fid.variables['x'][:]
+    y = nc_fid.variables['y'][:]
+    z = nc_fid.variables['z'][:]
+
+    # Find DEM boundary limits (to nearest 100)
+    x_lim = [np.ceil(min(x) / 100) * 100, np.floor(max(x) / 100) * 100]
+    y_lim = [np.ceil(min(y) / 100) * 100, np.floor(max(y) / 100) * 100]
+
+    # Arrange DEM tile corners
+    x_right_corner = np.arange(x_lim[1], x_lim[0] + tile_size, -unique_size - overlap).tolist()
+    y_bottom_corner = np.arange(y_lim[0], y_lim[1] - tile_size, unique_size + overlap).tolist()
+    # Combine to create the tile grid
+    [A, B] = np.meshgrid(x_right_corner, y_bottom_corner)
+    C = np.concatenate((A.T, B.T), axis=1)
+    bottom_right_corner = np.reshape(C, (int(C.size/2), 2), order='F')
+
+    # Tile boundaries:
+    left_bound  = bottom_right_corner[:, 0] - tile_size
+    right_bound = bottom_right_corner[:, 0]
+    lower_bound = bottom_right_corner[:, 1]
+    upper_bound = bottom_right_corner[:, 1] + tile_size
+
+    # Check the data
+    assert all(left_bound >= x_lim[0]), "Tiles violate min X bound."
+    assert all(right_bound <= x_lim[1]), "Tiles violate max X bound."
+    assert all(lower_bound >= y_lim[0]), "Tiles violate min Y bound."
+    assert all(upper_bound <= y_lim[1]), "Tiles violate max Y bound."
+
+    # Tile centers:
+    nTiles = len(bottom_right_corner)
+    x_tile_centers = (left_bound + right_bound) / 2
+    y_tile_centers = (lower_bound + upper_bound) / 2
+
+    P = Proj(proj='utm', zone=utm_zone, ellps='WGS84', preserve_units=True)
+    lon_tile_centers, lat_tile_centers = P(x_tile_centers, y_tile_centers, inverse=True)
+
+    # Log tile centers
+    tilesCenterFID = open(out_path + 'tile_centers.csv', 'w')
```

Comment:
@braanan nit: consider using a context a manager i.e.:
```suggestion
    with open(out_path + 'tile_centers.csv', 'w') as tilesCenterFID:
```

Reply:
Also, `out_path` may not necessarily include a trailing slash. Consider using `os.path.join` or `pathlib.Path` instances instead of concatenating strings.